src/components/progress-indicator/bl-progress-indicator.stories.mdx

```diff
@@ -0,0 +1,102 @@
+    },
+  }}
+/>
+
+export const ProgressIndicatorTemplate = (args) => html`
+<bl-progress-indicator
+    size='${ifDefined(args.size)}'
+    max='${ifDefined(args.max)}'
+    value='${ifDefined(args.value)}'
+    failed='${ifDefined(args.failed)}'
+    style='${ifDefined(args.styles ? styleMap(args.styles) : undefined)}'></bl-progress-indicator>`
+
+export const FailedTemplate = (args) => html`
+<div>
+<p style='color: var(--bl-color-danger); margin: var(--bl-size-3xs)'>Upload Failed - Image must not be larger than 3mb.</p>${ProgressIndicatorTemplate({ value:'100', failed:true, ...args })}
+</div>`;
+
+export const WithMaxTemplate = (args) => html`
+<div>
+<p style='color: var(--bl-color-content-secondary); margin: var(--bl-size-3xs)'> Completed Tasks: <b style='color: var(--bl-color-success)'> 5/8 </b></p>${ProgressIndicatorTemplate({ max:'8', value: '5', ...args })}
+</div>`;
+
+
+export const SizeTemplate = (args) => html`
+${ProgressIndicatorTemplate({value:'45', size:'small', styles: { 'margin' : 'var(--bl-size-2xs)' },...args })}
+${ProgressIndicatorTemplate({value:'45', size:'medium', styles: { 'margin' : 'var(--bl-size-2xs)' }, ...args })}
+${ProgressIndicatorTemplate({value:'45', size:'large',styles: { 'margin' : 'var(--bl-size-2xs)' }, ...args })}
+`;
+
+# Progress Indicator
+
+A progress indicator provides feedback about the duration and progression of a process to indicate how long a user will be waiting.
+
+Progress indicator component used for a long operation or a process that can take a considerable or unknown amount of time. It visually shows the progression of a system operation such as downloading, uploading, loading data, submitting a form, or saving updates.
+
+
+## Basic Usage
+
+By default, the `max` is 100 and the progress indicator is evaluated over 100. So the `value` must be a valid floating point number between 0 and `max`, or between 0 and 100 if `max` is omitted.
+
+
+<Canvas>
+  <Story name="Basic Usage" args={{ value: '50' }}>
+    {ProgressIndicatorTemplate.bind({})}
+  </Story>
+</Canvas>
+
+You don't have to pass the `value` according to 100 percent. For example, if you have a total of 8 tasks and completed 5 tasks you can pass parameters like `max='8' value='5'`. The progress indicator will divide into 8 parts and 5 parts will be full.
+
+<Canvas>
+  <Story name="Usage With Max Value">
+    {WithMaxTemplate.bind({})}
+  </Story>
+</Canvas>
+
+## Progress Indicator Status
+The progress indicator appears in success mode (green) by default. But if you need to show fail status for example failed to upload or failed to complete tasks you can pass `failed` parameters.
+
+<Canvas>
+  <Story name="Progress Indicator Status">
+    {FailedTemplate.bind({})}
+  </Story>
+</Canvas>
+
+## Progress Indicator Sizes
+
+We have 3 sizes of progress indicator: `large`, `medium`, `small`. Default size is `medium`.
+
+<Canvas>
+  <Story name="Progress Indicator Sizes">
+    {SizeTemplate.bind({})}
+  </Story>
```

Comment:
Add every size as a separate story. Then you'll not need to manually add margin for them and visual tests will be more clear.

Reply:
When I add stories for every size don't show the progress indicators. Because storybook adds div every story and this div width is 0 (I guess it works its child's width). Progress indicator width is 100% it works by the parent. So the component doesn't appear in the storybook. Maybe we can use different canvases.

Reply:
So this doesn't work?:

```jsx
<Canvas>
   <Story name="Progress Indicator Small" args={{ size: 'small' }}>
     {ProgressIndicatorTemplate.bind({})}
   </Story>
   <Story name="Progress Indicator Medium" args={{ size: 'medium' }}>
     {ProgressIndicatorTemplate.bind({})}
   </Story>
   <Story name="Progress Indicator Large" args={{ size: 'large' }}>
     {ProgressIndicatorTemplate.bind({})}
   </Story>
</Canvas>
```